kubos/test/integration/integration_test.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+
+import json
+import kubos
+import logging
+import mock
+import os
+import sys
+import unittest
+
+from sets import Set
+from kubos.test.utils import KubosTestCase
+from kubos.utils.constants import GLOBAL_TARGET_PATH
+
+class CLIIntegrationTest(KubosTestCase):
+
+    def setUp(self):
+        super(CLIIntegrationTest, self).setUp()
+        self.proj_name = 'project-test'
+        self.proj_dir = os.path.join(self.base_dir, self.proj_name)
+        self.target_list = None #Defined at test runtime
+        self.first_arg = sys.argv[0]
+        sys.argv = [self.first_arg] #clear any excess command line arguhements
+        logging.error = mock.MagicMock()
+
+
+        #The tests are run in alphabetical order by default (hence the 1,2,3,etc..)
+        #TODO: Get a better way of running the test in order while still being readable
+    def test_1_update_source_check_versions(self):
+        self.run_command('update')
+        os.chdir(self.base_dir)
+        self.run_command('version')
+        self.run_command('versions')
+        self.run_command('use', '--branch', 'master')
+        self.run_command('version')
+        self.run_command('versions')
+
+
+    def test_2_rt_example(self):
+        self.run_command('init', self.proj_name)
+        os.chdir(self.proj_dir)
+        self.run_command('target')
+        rt_list, linux_list = self.get_target_lists()
+        for target in rt_list:
```

Comment:
✨ Great work on this @kyleparrott. The TDD gods are smiling on us.

One future refactoring goal (not for this PR) might be to explode these loops into more verbose & explicit tests, per target. Yeah you heard me :bowtie: . Python bromides notwithstanding, explicit really is better in tests.

For instance I think it would be more readable (and also easier to read test output for) an explicit test for a target like:

```python
def test_rt_stm32f407-disco-gcc_basic_commands(self):
    self.run_command('init', '-l', self.proj_name)
    os.chdir(self.proj_dir)
    self.run_command('target')
    self.run_command('target', 'stm32f407-disco-gcc')
    self.run_command('target')
    self.run_command('list')
    self.run_command('build')
    self.run_command('clean')
    os.chdir(self.base_dir)
```

This does have the effect of bloating up the test files, but that's actually been the trend in the TDD world over the last few years; code duplication is actually encouraged, since each test has the benefit of being mostly stand-alone and readable in isolation without much knowledge of surrounding code, and it also has the effect of tending to make the test output more granular and easier to pinpoint problems from the test output alone.

A couple of opinionated articles for reference:
http://blog.jayfields.com/2008/05/testing-duplicate-code-in-your-tests.html
http://evgenii.com/blog/code-duplication-in-unit-tests/